extensions/ql-vscode/src/query-history/query-history-manager.ts

```diff
@@ -1177,57 +1178,40 @@ export class QueryHistoryManager extends DisposableObject {
   }
 
   private async findOtherQueryToCompare(
-    singleItem: QueryHistoryInfo,
-    multiSelect: QueryHistoryInfo[],
+    fromItem: CompletedLocalQueryInfo,
+    allItemsSelected: CompletedLocalQueryInfo[],
   ): Promise<CompletedLocalQueryInfo | undefined> {
-    // Variant analyses cannot be compared
-    if (
-      singleItem.t !== "local" ||
-      multiSelect.some((s) => s.t !== "local") ||
-      !singleItem.completedQuery
-    ) {
-      return undefined;
-    }
-    const dbName = singleItem.initialInfo.databaseInfo.name;
+    const dbName = fromItem.initialInfo.databaseInfo.name;
 
     // if exactly 2 queries are selected, use those
-    if (multiSelect?.length === 2) {
-      // return the query that is not the first selected one
-      const otherQuery = (
-        singleItem === multiSelect[0] ? multiSelect[1] : multiSelect[0]
-      ) as LocalQueryInfo;
-      if (!otherQuery.completedQuery) {
-        throw new Error("Please select a completed query.");
-      }
-      if (!otherQuery.completedQuery.successful) {
-        throw new Error("Please select a successful query.");
-      }
-      if (otherQuery.initialInfo.databaseInfo.name !== dbName) {
+    if (allItemsSelected.length === 2) {
+      const otherItem =
+        fromItem === allItemsSelected[0]
+          ? allItemsSelected[1]
+          : allItemsSelected[0];
+      if (otherItem.initialInfo.databaseInfo.name !== dbName) {
         throw new Error("Query databases must be the same.");
       }
-      return otherQuery as CompletedLocalQueryInfo;
+      return otherItem;
     }
 
-    if (multiSelect?.length > 2) {
+    if (allItemsSelected.length > 2) {
       throw new Error("Please select no more than 2 queries.");
     }
 
     // otherwise, let the user choose
     const comparableQueryLabels = this.treeDataProvider.allHistory
+      .filter(this.isSuccessfulCompletedLocalQueryInfo)
       .filter(
-        (otherQuery) =>
-          otherQuery !== singleItem &&
-          otherQuery.t === "local" &&
-          otherQuery.completedQuery &&
-          otherQuery.completedQuery.successful &&
-          otherQuery.initialInfo.databaseInfo.name === dbName,
+        (otherItem) =>
+          otherItem !== fromItem &&
+          otherItem.initialInfo.databaseInfo.name === dbName,
```

Comment:
We dig for this database name in a few places. Could we assign `otherItem.initialInfo.databaseInfo.name` to a variable at the top to make this terser, the same way we do for `fromItem`?

e.g.  `otherItemDbName`

Reply:
We can't really assign to a variable, because we're operating on a different `otherItem` variable each time, even if they use the same name.

What we could do is create a `getDatabaseName` method, which will make things a bit less repetitive.